conf/test_route_control.py

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
+
+import sys
+import unittest
+from unittest.mock import MagicMock, Mock, patch
+
+from pyroute2 import IPDB  # type: ignore[import]
+
+sys.modules["pybess.bess"] = MagicMock()
+
+from conf.route_control import (
+    NeighborEntry,
+    RouteController,
+    RouteEntry,
+    fetch_mac,
+    mac_to_hex,
+    mac_to_int,
+    validate_ipv4,
+)
+
+
+class BessControllerMock(object):
+    """Mock of BessController to avoid using BESS from pybess.bess"""
+
+    def __init__(self):
+        pass
+
+    def _get_bess(self, *args, **kwargs) -> None:
+        pass
+
+    def add_route_to_module(self, *args, **kwargs) -> None:
+        pass
+
+    def delete_module_route_entry(self, *args, **kwargs) -> None:
+        pass
+
+    def create_module(self, *args, **kwargs) -> None:
+        pass
+
+    def delete_module(self, *args, **kwargs) -> None:
+        pass
+
+    def link_modules(self, *args, **kwargs) -> None:
+        pass
+
+
+@patch("conf.route_control.BessController", BessControllerMock)
+class TestUtilityFunctions(unittest.TestCase):
+    """Tests utility functions in route_control.py."""
+
+    def test_given_valid_ip_when_validate_ipv4_then_returns_true(self):
+        self.assertTrue(validate_ipv4("192.168.1.1"))
+
+    def test_given_invalid_ip_when_validate_ipv4_then_returns_false(self):
+        self.assertFalse(validate_ipv4("192.168.300.1"))
+
+    def test_given_invalid_ip_when_validate_ipv4_then_returns_false(self):
```

Comment:
Hi @saltiyazan, I am working on replacing the IPDB for the `NDB` and while doing so, I noticed that this test has the exact same name as the previous definition/function. Should it be IPv6 instead of IPv4?
```diff
-    def test_given_invalid_ip_when_validate_ipv4_then_returns_false(self):
+    def test_given_invalid_ip_when_validate_ipv6_then_returns_false(self):
```

Reply:
Hi @gab-arrobo 
It seems like this should have been one test case asserting on different values.